Source/WebCore/html/HTMLButtonElement.cpp

```diff
@@ -155,10 +155,15 @@ void HTMLButtonElement::defaultEventHandler(Event& event)
 
             if (m_type == SUBMIT || m_type == RESET)
                 event.setDefaultHandled();
-        } else if (invokeTargetElement()) {
-            handleInvokeAction();
-        } else
-            handlePopoverTargetAction();
+        }
+
+        if (!(protectedForm && m_type == SUBMIT)) {
+            if (invokeTargetElement())
+                handleInvokeAction();
+            else
+                handlePopoverTargetAction();
```

Comment:
Do we need to call `setDefaultHandled` for any edges cases? Can we add tests that cover that explicitly?

Reply:
Included as part of follow up bug